Move more ivars from WKWebViewConfiguration to API::PageConfiguration

achristensen07 opened the pull request:
#### f3a7825e8d6e739ddbcda9db8f7b1a1c992843c6
<pre>
Move more ivars from WKWebViewConfiguration to API::PageConfiguration
<a href="https://bugs.webkit.org/show_bug.cgi?id=271438">https://bugs.webkit.org/show_bug.cgi?id=271438</a>

Reviewed by Charlie Wolfe.

* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::attachmentFileWrapperClass const):
(API::PageConfiguration::setAttachmentFileWrapperClass):
(API::PageConfiguration::additionalSupportedImageTypes const):
(API::PageConfiguration::setAdditionalSupportedImageTypes):
(API::PageConfiguration::applePayEnabled const):
(API::PageConfiguration::setApplePayEnabled):
(API::PageConfiguration::appHighlightsEnabled const):
(API::PageConfiguration::setAppHighlightsEnabled):
(API::PageConfiguration::groupIdentifier const):
(API::PageConfiguration::setGroupIdentifier):
(API::PageConfiguration::mediaContentTypesRequiringHardwareSupport const):
(API::PageConfiguration::setMediaContentTypesRequiringHardwareSupport):
(API::PageConfiguration::applicationNameForUserAgent const):
(API::PageConfiguration::setApplicationNameForUserAgent):
(API::PageConfiguration::sampledPageTopColorMaxDifference const):
(API::PageConfiguration::setSampledPageTopColorMaxDifference):
(API::PageConfiguration::sampledPageTopColorMinHeight const):
(API::PageConfiguration::setSampledPageTopColorMinHeight):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration init]):
(-[WKWebViewConfiguration encodeWithCoder:]):
(-[WKWebViewConfiguration copyWithZone:]):
(-[WKWebViewConfiguration _applicationNameForDesktopUserAgent]):
(-[WKWebViewConfiguration applicationNameForUserAgent]):
(-[WKWebViewConfiguration setApplicationNameForUserAgent:]):
(-[WKWebViewConfiguration _groupIdentifier]):
(-[WKWebViewConfiguration _setGroupIdentifier:]):
(-[WKWebViewConfiguration _attachmentFileWrapperClass]):
(-[WKWebViewConfiguration _setAttachmentFileWrapperClass:]):
(-[WKWebViewConfiguration _isControlledByAutomation]):
(-[WKWebViewConfiguration _setControlledByAutomation:]):
(-[WKWebViewConfiguration _applePayEnabled]):
(-[WKWebViewConfiguration _setApplePayEnabled:]):
(-[WKWebViewConfiguration _mediaContentTypesRequiringHardwareSupport]):
(-[WKWebViewConfiguration _setMediaContentTypesRequiringHardwareSupport:]):
(-[WKWebViewConfiguration _additionalSupportedImageTypes]):
(-[WKWebViewConfiguration _setAdditionalSupportedImageTypes:]):
(-[WKWebViewConfiguration _setAppHighlightsEnabled:]):
(-[WKWebViewConfiguration _appHighlightsEnabled]):
(-[WKWebViewConfiguration _setSampledPageTopColorMaxDifference:]):
(-[WKWebViewConfiguration _sampledPageTopColorMaxDifference]):
(-[WKWebViewConfiguration _setSampledPageTopColorMinHeight:]):
(-[WKWebViewConfiguration _sampledPageTopColorMinHeight]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):

Canonical link: <a href="https://commits.webkit.org/276532@main">https://commits.webkit.org/276532@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f04413a13dce4cca1183fa7a5314fc39d4fdff56

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44905 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/24004 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47395 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47559 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40910 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/28065 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21401 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36862 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45483 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/21055 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38662 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17951 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/18467 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2952 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/41132 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/40101 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49231 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19873 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16412 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/43857 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/21192 "Built successfully") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/42619 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/21530 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6233 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20866 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
```diff
@@ -209,8 +178,8 @@ - (void)encodeWithCoder:(NSCoder *)coder
 
-    if (_applicationNameForUserAgent)
-        [coder encodeObject:self.applicationNameForUserAgent forKey:@"applicationNameForUserAgent"];
+    if (NSString *name = self.applicationNameForUserAgent)
```
Isn't this a behavior change? Previously this was checking if `_applicationNameForUserAgent` was `std::nullopt`, now it's checking if the `NSString *` is nil.

achristensen07 replied:
It could be.  I'll keep the null and nullopt states distinct in this PR.

charliewolfe commented on Source/WebKit/UIProcess/API/APIPageConfiguration.h:
```diff
@@ -414,7 +443,8 @@ class PageConfiguration : public ObjectImpl<Object::Type::PageConfiguration> {
 #if PLATFORM(COCOA)
-        WTF::Vector<WTF::String> additionalSupportedImageTypes { };
+        RetainPtr<ClassStructPtr> attachmentFileWrapperClass;
+        std::optional<WTF::Vector<WTF::String>> additionalSupportedImageTypes { };
```
```suggestion
        std::optional<WTF::Vector<WTF::String>> additionalSupportedImageTypes;
```

charliewolfe commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
```diff
@@ -152,19 +134,6 @@ - (instancetype)init
-#endif
-
-    _sampledPageTopColorMaxDifference = DEFAULT_VALUE_FOR_SampledPageTopColorMaxDifference;
-    _sampledPageTopColorMinHeight = DEFAULT_VALUE_FOR_SampledPageTopColorMinHeight;
```
`#import "WebPreferencesDefaultValues.h"` can probably be removed from this file now.

achristensen07 committed:
Move more ivars from WKWebViewConfiguration to API::PageConfiguration
https://bugs.webkit.org/show_bug.cgi?id=271438

Reviewed by Charlie Wolfe.

* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::attachmentFileWrapperClass const):
(API::PageConfiguration::setAttachmentFileWrapperClass):
(API::PageConfiguration::additionalSupportedImageTypes const):
(API::PageConfiguration::setAdditionalSupportedImageTypes):
(API::PageConfiguration::applePayEnabled const):
(API::PageConfiguration::setApplePayEnabled):
(API::PageConfiguration::appHighlightsEnabled const):
(API::PageConfiguration::setAppHighlightsEnabled):
(API::PageConfiguration::groupIdentifier const):
(API::PageConfiguration::setGroupIdentifier):
(API::PageConfiguration::mediaContentTypesRequiringHardwareSupport const):
(API::PageConfiguration::setMediaContentTypesRequiringHardwareSupport):
(API::PageConfiguration::applicationNameForUserAgent const):
(API::PageConfiguration::setApplicationNameForUserAgent):
(API::PageConfiguration::sampledPageTopColorMaxDifference const):
(API::PageConfiguration::setSampledPageTopColorMaxDifference):
(API::PageConfiguration::sampledPageTopColorMinHeight const):
(API::PageConfiguration::setSampledPageTopColorMinHeight):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration init]):
(-[WKWebViewConfiguration encodeWithCoder:]):
(-[WKWebViewConfiguration copyWithZone:]):
(-[WKWebViewConfiguration _applicationNameForDesktopUserAgent]):
(-[WKWebViewConfiguration applicationNameForUserAgent]):
(-[WKWebViewConfiguration setApplicationNameForUserAgent:]):
(-[WKWebViewConfiguration _groupIdentifier]):
(-[WKWebViewConfiguration _setGroupIdentifier:]):
(-[WKWebViewConfiguration _attachmentFileWrapperClass]):
(-[WKWebViewConfiguration _setAttachmentFileWrapperClass:]):
(-[WKWebViewConfiguration _isControlledByAutomation]):
(-[WKWebViewConfiguration _setControlledByAutomation:]):
(-[WKWebViewConfiguration _applePayEnabled]):
(-[WKWebViewConfiguration _setApplePayEnabled:]):
(-[WKWebViewConfiguration _mediaContentTypesRequiringHardwareSupport]):
(-[WKWebViewConfiguration _setMediaContentTypesRequiringHardwareSupport:]):
(-[WKWebViewConfiguration _additionalSupportedImageTypes]):
(-[WKWebViewConfiguration _setAdditionalSupportedImageTypes:]):
(-[WKWebViewConfiguration _setAppHighlightsEnabled:]):
(-[WKWebViewConfiguration _appHighlightsEnabled]):
(-[WKWebViewConfiguration _setSampledPageTopColorMaxDifference:]):
(-[WKWebViewConfiguration _sampledPageTopColorMaxDifference]):
(-[WKWebViewConfiguration _setSampledPageTopColorMinHeight:]):
(-[WKWebViewConfiguration _sampledPageTopColorMinHeight]):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::creationParameters):

Canonical link: https://commits.webkit.org/276532@main
